packages/next/build/webpack/plugins/flight-types-plugin.ts

```diff
@@ -93,19 +94,29 @@ async function collectNamedSlots(layoutPath: string) {
 
 export class FlightTypesPlugin {
   dir: string
+  distDir: string
   appDir: string
   dev: boolean
   isEdgeServer: boolean
 
   constructor(options: Options) {
     this.dir = options.dir
+    this.distDir = options.distDir
     this.appDir = options.appDir
     this.dev = options.dev
     this.isEdgeServer = options.isEdgeServer
   }
 
   apply(compiler: webpack.Compiler) {
-    const assetPrefix = this.dev ? '..' : this.isEdgeServer ? '..' : '../..'
+    // From dist root to project root
+    const distDirRelative = path.relative(this.distDir + '/..', '.')
```

Comment:
for path normalization safety
```suggestion
    const distDirRelative = path.relative(this.distDir, '..', '.')
```

Reply:
They're not the same I believe.